allennlp/commands/checklist.py

```diff
@@ -2,203 +2,40 @@
-        )
-
-    file_path = args.checklist_suite
-
-    task_suite_args = args.task_suite_args.strip()
-    if len(task_suite_args) <= 0:
-        task_suite_args = {}
-    else:
-        task_suite_args = json.loads(task_suite_args)
-
-    return TaskSuite.constructor(
-        name=suite_name,
-        suite_file=file_path,
-        extra_args=task_suite_args,
-    )
-
-
-class _CheckListManager:
-    def __init__(
-        self,
-        task_suite: TaskSuite,
-        predictor: Predictor,
-        capabilities: Optional[List[str]] = None,
-        max_examples: Optional[int] = None,
-        output_file: Optional[str] = None,
-        print_summary_args: Optional[Dict[str, Any]] = None,
-    ) -> None:
-        self._task_suite = task_suite
-        self._predictor = predictor
-        self._capabilities = capabilities
-        self._max_examples = max_examples
-        self._output_file = None if output_file is None else open(output_file, "w")
-        self._print_summary_args = print_summary_args or {}
-
-        if capabilities:
-            self._print_summary_args["capabilities"] = capabilities
-
-    def run(self) -> None:
-        self._task_suite.run(
-            self._predictor, capabilities=self._capabilities, max_examples=self._max_examples
-        )
-
-        # We pass in an IO object.
-        output_file = self._output_file or sys.stdout
-        self._task_suite.summary(file=output_file, **self._print_summary_args)
-
-        # If `_output_file` was None, there would be nothing to close.
-        if self._output_file is not None:
-            self._output_file.close()
-
-
-def _run_suite(args: argparse.Namespace) -> None:
-
-    task_suite = _get_task_suite(args)
-    predictor = _get_predictor(args)
-
-    print_summary_args = args.print_summary_args.strip()
-    if len(print_summary_args) <= 0:
-        print_summary_args = {}
-    else:
-        print_summary_args = json.loads(print_summary_args)
-
-    capabilities = args.capabilities
-    max_examples = args.max_examples
-
-    manager = _CheckListManager(
-        task_suite,
-        predictor,
-        capabilities,
-        max_examples,
-        args.output_file,
-        print_summary_args,
-    )
-    manager.run()
+    # create dummy command that tells users how to
+    # install the necessary dependency
+
+    def _dummy_output(args: argparse.Namespace):
+        logger.info(
+            "The checklist integration of allennlp is optional; if you're using conda, "
+            "it can be installed with `conda install allennlp-checklist`, "
+            "otherwise use `pip install allennlp[checklist]`."
+        )
+
+    # need to work around https://github.com/python/mypy/issues/1153
+    @Subcommand.register("checklist")
+    class CheckList(Subcommand):  # type: ignore
+        def add_subparser(self, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+            description = """Dummy command because checklist is not installed."""
```

Comment:
Can we add the installation info here, so that it shows up if someone does `allennlp checklist --help`?

Reply:
Hey @AkshitaB, thanks for taking a look.

That is indeed the intention, i.e. that any call to `allennlp checklist` (when `checklist` is missing) will print the installation instructions (see `_dummy_output` above - name can be improved). I don't know the `SubCommand` infrastructure well enough, but happy to adapt as necessary.

Reply:
@h-vetinari I think changing the description string to the `_dummy_output` message should do the trick.

Reply:
My idea was to print the installation tip unconditionally for any invocation of `allennlp checklist`, not just for `allennlp checklist --help`. Does that make sense? I'm happy to change it though if desired.